src/app/model/HelperFunctions.ts

```diff
@@ -5,3 +5,7 @@ export function isDefined<T>(x: T | undefined): x is T {
 export function isNotDefined<T>(x: T | undefined): x is undefined {
   return !isDefined(x);
 }
+
+export function hasOccurred<T>(x: T | undefined): x is T {
```

Comment:
Zostawiłbym to jako metodę prywatną w ChessEngine. Teraz jak jest publiczna, to nie jest już szczegółem implementacyjnym i musiałaby być testowana i utrzymywana osobno :)